taprpc/authmailboxrpc/mailbox.proto

```diff
@@ -0,0 +1,229 @@
+    repeated MailboxMessage messages = 1;
+}
+
+// Request message for the SendMessage RPC.
+message SendMessageRequest {
+    // The public key identifier of the intended receiver (ReceiverID), encoded
+    // as the raw bytes of the compressed public key.
+    bytes receiver_id = 1;
+
+    // The ECIES encrypted message payload.
+    bytes encrypted_payload = 2;
+
+    // A proof of the sender's identity, typically a Bitcoin transaction or
+    // another proof that work was done to create the message.
+    oneof proof {
+        // The Bitcoin Merkle Inclusion Proof used as the sender's
+        // authentication. The server MUST perform full validation of this
+        // proof:
+        // 1. Verify claimed_outpoint.txid_hex matches hash(raw_tx_data).
+        // 2. Verify claimed_outpoint.index is valid for the transaction.
+        // 3. Verify merkle_proof connects the transaction hash to the
+        //    raw_block_header_data's Merkle root.
+        // 4. Verify block_header validity (e.g., PoW, potentially chain
+        //    context).
+        // 5. Ensure the claimed_outpoint has not been used previously (check
+        //    used_proofs table).
+        BitcoinMerkleInclusionProof tx_proof = 4;
+    }
+
+    // The block height after which this message may be considered expired.
+    uint32 expiry_block_height = 5;
+}
+
+// Response message for the SendMessage RPC.
+message SendMessageResponse {
+    // The unique ID assigned to the stored message by the server.
+    uint64 message_id = 1;
+}
+
+// Wrapper message for requests sent FROM the client TO the server during the
+// ReceiveMessages stream.
+message ReceiveMessagesRequest {
+    oneof request_type {
+        InitReceive init = 1;
+        AuthSignature auth_sig = 2;
+    }
+}
+
+// Carries the initial parameters from the client to start receiving messages.
+message InitReceive {
+    bytes receiver_id = 1;
+    uint64 start_message_id_exclusive = 2;
+    uint32 start_block_height_inclusive = 3;
+    int64 start_timestamp_exclusive = 4;
+}
+
+// Carries the client's signature in response to the server's challenge.
+message AuthSignature {
+    bytes signature = 1;
+}
+
+// Wrapper message for responses sent FROM the server TO the client
+// during the ReceiveMessages stream.
+message ReceiveMessagesResponse {
+    oneof response_type {
+        Challenge challenge = 1;
+        bool auth_success = 2;
+        MailboxMessages messages = 3;
+        ReceiveError error = 4;
+        EndOfStream eos = 5;
+    }
+}
+
+// Carries the challenge hash sent by the server to the client.
+message Challenge {
+    bytes challenge_hash = 1;
+}
+
+// Represents a structured error encountered during the ReceiveMessages stream.
+message ReceiveError {
+    enum Code {
+        CODE_UNSPECIFIED = 0;
+        INVALID_RECEIVER_ID = 1;
+        INVALID_SIGNATURE = 2;
+        AUTH_TIMEOUT = 3;
+        INTERNAL_ERROR = 4;
+        TEMPORARILY_UNAVAILABLE = 5;
```

Comment:
I think we should add doc comments for these enum values and also for the fields in the other messages.

Reply:
Actually, we're not using these error codes at the moment, so I removed them instead.